pgrx/src/pg_catalog.rs

```diff
@@ -0,0 +1,866 @@
+                            $p0_name.into_datum()?,
+                            $p1_name.into_datum()?,
+                            $p2_name.into_datum()?,
+                            $p3_name.into_datum()?,
+                        );
+                        let inner = std::ptr::NonNull::new(entry);
+                        Some([<$catalog:camel Search>] { inner, cache_id })
+                    }
+                }
+                pub fn [<search _list_ $cache _1>]($p0_name: $p0_type) -> Option<[<$catalog:camel SearchList>]> {
+                    unsafe {
+                        use pgrx::datum::IntoDatum;
+                        let cache_id = pgrx::pg_sys::[<SysCacheIdentifier_ $cache:upper>] as i32;
+                        let entry = pgrx::pg_sys::SearchSysCacheList(
+                            cache_id,
+                            1,
+                            $p0_name.into_datum()?,
+                            0.into(),
+                            0.into(),
+                        );
+                        let inner = std::ptr::NonNull::new(entry).unwrap();
+                        Some([<$catalog:camel SearchList>] { inner, cache_id })
+                    }
+                }
+                pub fn [<search _list_ $cache _2>]($p0_name: $p0_type, $p1_name: $p1_type) -> Option<[<$catalog:camel SearchList>]> {
+                    unsafe {
+                        use pgrx::datum::IntoDatum;
+                        let cache_id = pgrx::pg_sys::[<SysCacheIdentifier_ $cache:upper>] as i32;
+                        let entry = pgrx::pg_sys::SearchSysCacheList(
+                            cache_id,
+                            2,
+                            $p0_name.into_datum()?,
+                            $p1_name.into_datum()?,
+                            0.into(),
+                        );
+                        let inner = std::ptr::NonNull::new(entry).unwrap();
+                        Some([<$catalog:camel SearchList>] { inner, cache_id })
+                    }
+                }
+                pub fn [<search _list_ $cache _3>]($p0_name: $p0_type, $p1_name: $p1_type, $p2_name: $p2_type) -> Option<[<$catalog:camel SearchList>]> {
+                    unsafe {
+                        use pgrx::datum::IntoDatum;
+                        let cache_id = pgrx::pg_sys::[<SysCacheIdentifier_ $cache:upper>] as i32;
+                        let entry = pgrx::pg_sys::SearchSysCacheList(
+                            cache_id,
+                            3,
+                            $p0_name.into_datum()?,
+                            $p1_name.into_datum()?,
+                            $p2_name.into_datum()?,
+                        );
+                        let inner = std::ptr::NonNull::new(entry).unwrap();
+                        Some([<$catalog:camel SearchList>] { inner, cache_id })
+                    }
+                }
+            }
+        }
+    };
+}
+
+macro_rules! defines {
```

Comment:
Defines what?

Reply:
Renamed to `define_catalogs_and_caches`.